src/server/tiering/op_manager.cc

```diff
@@ -102,11 +102,21 @@ void OpManager::ProcessStashed(EntryId id, unsigned version, DiskSegment segment
 }
 
 void OpManager::ProcessRead(size_t offset, std::string_view value) {
+  util::FiberAtomicGuard guard;  // atomically update items, no in-between states should be possible
   ReadOp* info = &pending_reads_.at(offset);
 
+  // Reorder base read (offset 0) to be last, so reads for defragmentation are handled last
+  for (size_t i = 0; i + 1 < info->key_ops.size(); i++) {
+    if (info->key_ops[i].segment.offset % kPageSize == 0) {
+      std::swap(info->key_ops[i], info->key_ops.back());
+      break;
+    }
+  }
+
   bool deleting_full = false;
   std::string key_value;
-  for (auto& ko : info->key_ops) {
+  for (size_t i = 0; i < info->key_ops.size(); i++) {  // more items can be read while iterating
```

Comment:
what does it mean `more items can be read while iterating` ?

Reply:
When we call ReportDelete, it might issue a read for the whole page to trigger defragmentation, which will be added to this list

Reply:
please add a comment:
// Note that some function in the loop may append items to `info->key_ops` during the traversal.